rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/InterpretedLanguageContributions.java

```diff
@@ -234,16 +228,51 @@ private static CompletableFuture<Boolean> nonNull(CompletableFuture<?> x) {
         return x.thenApply(Objects::nonNull);
     }
 
-    private static CompletableFuture<Boolean> summaryConfigLookup(CompletableFuture<IConstructor> conf, String parameter) {
-        return conf.thenApply( d -> {
-            if (d == null) {
-                return false;
+    private static CompletableFuture<SummaryConfig> scheduledSummaryConfig(CompletableFuture<ISet> contributions, String summarizer) {
+        return contributions.thenApply(c -> {
+            var constructor = getContribution(c, summarizer);
+            if (constructor != null) {
+                return new SummaryConfig(
+                    isTrue(constructor, "providesDocumentation"),
```

Comment:
small style breach here since every string constant has been factored.

Reply:
good, also refactored that :+1: